<?xml version="1.0"?>
<launch>
  	<!-- name for the camera -->
  	<arg name="camera_name" default="r50" />
  	<!-- frame_id for the camera -->
  	<arg name="frame_id" default="camera" /> 
  	<arg name="distance_max" default="5000"/>

   	<!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
   	<group ns="$(arg camera_name)">
	    <node pkg="depthsense_publisher" type="depthsense_publisher" name="$(arg camera_name)_stream" output="screen"> 
	    	<param name="camera_name" type="string" value="$(arg camera_name)" />
	        <param name="frame_id" type="string" value="$(arg frame_id)" />
	        <param name="distance_max" type="int" value="$(arg distance_max)" />			
	    </node>

	    <node name="rviz" pkg="rviz" type="rviz" />

	    <node name="image_viewer" pkg="image_view" type="image_view">
	    	<remap from="image" to="image_raw" />
	    </node>
	    <node name="depth_viewer" pkg="image_view" type="image_view">
	    	<remap from="image" to="depth_raw" />
	    </node>			
	</group>

</launch>
